src/brogue/Monsters.c

```diff
@@ -230,6 +230,30 @@ boolean canDirectlySeeMonster(creature *monst) {
     return false;
 }
 
+// Centralize the logic of whether an attacker can/will attack the defender
+boolean canAttack( creature *attacker, creature *defender ) {
```

Comment:
We should decide what this means precisely. For example, this check includes that the defender's location is known, but it's obviously possible to attack completely invisible enemies.

It probably means something more like "does the attack know it will attack defender if it moves into its cell?"